java/src/jmri/jmrit/display/layoutEditor/LayoutEditor.java

```diff
@@ -2021,14 +2027,39 @@ protected void targetWindowClosingEvent(java.awt.event.WindowEvent e) {
      * @param inValidateMode boolean: if true, valid text == green, invalid text
      *                       == red background; if false, valid text == green,
      *                       invalid text == yellow background
-     * @param inEnable       boolean to enable / disable the JmriBeanComboBoxes
+     * @param inEnable       boolean to enable / disable the JmriBeanComboBox
      */
     public void setupComboBox(JmriBeanComboBox inComboBox, boolean inValidateMode, boolean inEnable) {
         inComboBox.setEnabled(inEnable);
         inComboBox.setEditable(true);
         inComboBox.setValidateMode(inValidateMode);
         inComboBox.setText("");
+        int c = Math.max(8, inComboBox.getItemCount());
```

Comment:
The while loop makes c an even division of the number of items in the dropdown list less than or equal to 64 items (an approximation of how many items would fit onscreen. I could probably calculate the number of items that fit (vertically) onscreen but that's more work than is really necessary given what it's used for.).